modules/core/shared/src/main/scala/io/circe/JsonNumber.scala

```diff
@@ -236,6 +237,18 @@ private[circe] final case class JsonFloat(value: Float) extends JsonNumber {
  */
 object JsonNumber {
 
+  /**
+  * Constant pool of integer numbers between -128 to 127, similar to the JVM's constant pool for integers.
+  */
+  private val jsonLongConstantPool: ArraySeq[JsonLong] =
+    ArraySeq.tabulate(256)(x => JsonLong((x - 128).toLong))
+
+  private[circe] final def fromLong(value: Long): JsonLong = {
+    val idx = value + 128
+    if (idx >= 0 && idx < jsonLongConstantPool.size) jsonLongConstantPool.apply(idx.toInt)
```

Comment:
```suggestion
    if (idx >= 0 && idx < jsonLongConstantPool.length) jsonLongConstantPool.apply(idx.toInt)
```

Reply:
Decompile after these changes and it should avoid a couple method calls from your previous result :)

Reply:
Indeed, much cleaner:
```java
    public final JsonLong fromLong(long value) {
        long idx = value + 128L;
        if (idx >= 0L && idx < (long)jsonLongConstantPool.length) {
            return jsonLongConstantPool[(int)idx];
        }
        return new JsonLong(value);
    }
```